runtime/common/src/integration_tests.rs

```diff
@@ -298,171 +300,176 @@ fn last_event() -> Event {
 	System::events().pop().expect("Event expected").event
 }
 
+// Runs an end to end test of the auction, crowdloan, slots, and onboarding process over varying
+// lease period offsets.
 #[test]
 fn basic_end_to_end_works() {
-	new_test_ext().execute_with(|| {
-		let para_1 = LOWEST_PUBLIC_ID;
-		let para_2 = LOWEST_PUBLIC_ID + 1;
-		assert!(System::block_number().is_one());
-		// User 1 and 2 will own parachains
-		Balances::make_free_balance_be(&1, 1_000_000_000);
-		Balances::make_free_balance_be(&2, 1_000_000_000);
-		// First register 2 parathreads
-		let genesis_head = Registrar::worst_head_data();
-		let validation_code = Registrar::worst_validation_code();
-		assert_ok!(Registrar::reserve(Origin::signed(1)));
-		assert_ok!(Registrar::register(
-			Origin::signed(1),
-			ParaId::from(para_1),
-			genesis_head.clone(),
-			validation_code.clone(),
-		));
-		assert_ok!(Registrar::reserve(Origin::signed(2)));
-		assert_ok!(Registrar::register(
-			Origin::signed(2),
-			ParaId::from(2001),
-			genesis_head,
-			validation_code,
-		));
-
-		// Paras should be onboarding
-		assert_eq!(Paras::lifecycle(ParaId::from(para_1)), Some(ParaLifecycle::Onboarding));
-		assert_eq!(Paras::lifecycle(ParaId::from(para_2)), Some(ParaLifecycle::Onboarding));
-
-		// Start a new auction in the future
-		let duration = 99u32;
-		let lease_period_index_start = 4u32;
-		assert_ok!(Auctions::new_auction(Origin::root(), duration, lease_period_index_start));
-
-		// 2 sessions later they are parathreads
-		run_to_session(2);
-		assert_eq!(Paras::lifecycle(ParaId::from(para_1)), Some(ParaLifecycle::Parathread));
-		assert_eq!(Paras::lifecycle(ParaId::from(para_2)), Some(ParaLifecycle::Parathread));
-
-		// Para 1 will bid directly for slot 1, 2
-		// Open a crowdloan for Para 2 for slot 3, 4
-		assert_ok!(Crowdloan::create(
-			Origin::signed(2),
-			ParaId::from(para_2),
-			1_000,                        // Cap
-			lease_period_index_start + 2, // First Slot
-			lease_period_index_start + 3, // Last Slot
-			200,                          // Block End
-			None,
-		));
-		let crowdloan_account = Crowdloan::fund_account_id(ParaId::from(para_2));
-
-		// Auction ending begins on block 100, so we make a bid before then.
-		run_to_block(90);
-
-		Balances::make_free_balance_be(&10, 1_000_000_000);
-		Balances::make_free_balance_be(&20, 1_000_000_000);
+	for offset in [0u32, 50, 100, 200].iter() {
```

Comment:
This test is basically unchanged except introducing `offset` into the timeline of the setup.